packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,376 @@
+ */
+
+import * as path from "path";
+import { readdirSync } from "fs";
+import { strict as assert } from "assert";
+import {
+	AcceptanceCondition,
+	createWeightedAsyncGenerator as createWeightedGenerator,
+	AsyncGenerator as Generator,
+	takeAsync as take,
+} from "@fluid-internal/stochastic-test-utils";
+import {
+	createDDSFuzzSuite,
+	DDSFuzzModel,
+	DDSFuzzHarnessEvents,
+} from "@fluid-internal/test-dds-utils";
+import { PropertySet } from "@fluidframework/merge-tree";
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { IntervalCollection, SequenceInterval } from "../intervalCollection";
+import { SharedStringFactory } from "../sequenceFactory";
+import { SharedString } from "../sharedString";
+import {
+	appendAddIntervalToRevertibles,
+	// appendChangeIntervalToRevertibles,
+	// appendDeleteIntervalToRevertibles,
+	appendIntervalPropertyChangedToRevertibles,
+	appendSharedStringDeltaToRevertibles,
+	SharedStringRevertible,
+} from "../revertibles";
+import { assertEquivalentSharedStrings } from "./intervalUtils";
+import {
+	Operation,
+	RangeSpec,
+	AddText,
+	RemoveRange,
+	AddInterval,
+	DeleteInterval,
+	ChangeInterval,
+	ChangeProperties,
+	FuzzTestState,
+	makeRevertibleReducer,
+	RevertSharedStringRevertibles,
+	RevertOperation,
+} from "./intervalCollection.fuzzUtils";
+import { minimizeTestFromFailureFile } from "./intervalCollection.fuzzMinimization";
+
+// Note: none of these options are currently exercised, since the fuzz test fails with pretty much
+// any configuration due to known bugs. Once shared interval collections are in a better state these
+// should be revisited.
+interface OperationGenerationConfig {
+	/**
+	 * Maximum length of the SharedString (locally) before no further AddText operations are generated.
+	 * Note due to concurency, during test execution the actual length of the string may exceed this.
+	 */
+	maxStringLength?: number;
+	/**
+	 * Maximum number of intervals (locally) before no further AddInterval operations are generated.
+	 * Note due to concurency, during test execution the actual number of intervals may exceed this.
+	 */
+	maxIntervals?: number;
+	maxInsertLength?: number;
+	intervalCollectionNamePool?: string[];
+	propertyNamePool?: string[];
+	validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+	maxStringLength: 1000,
+	maxIntervals: 100,
+	maxInsertLength: 10,
+	intervalCollectionNamePool: ["comments"],
+	propertyNamePool: ["prop1", "prop2", "prop3"],
+	validateInterval: 100,
+};
+
+const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
+const revertibles: SharedStringRevertible[] = [];
+
+emitter.on("clientCreate", (client) => {
+	const string = client.channel as SharedString;
+	const labels = string.getIntervalCollectionLabels();
+	Array.from(labels).forEach((label) => {
+		const collection = string.getIntervalCollection(label);
+		collection.on("addInterval", (interval, local, op) => {
```

Comment:
i think you probably also want this subscription logic attached to whenever the string emits a `"createIntervalCollection"`.

Reply:
(right now, you only subscribe to collections that exist at string creation time, not ones created while the client is collaborating)

Reply:
unresolving--looks like now you have the logic in the event but not for all collections that might exist at load time :). You want both. I'd do something like

```typescript
const subscribeToIntervalCollectionChanges = (label: string) => {
    // ...
}

for (const label of channel.getIntervalCollectionLabels()) {
    subscribeToIntervalCollectionChanges(label);
}

channel.on("createIntervalCollection", subscribeToIntervalCollectionChanges);
```